recipes/apr/all/conanfile.py

```diff
@@ -67,6 +62,13 @@ def _configure_cmake(self):
     def _configure_autotools(self):
         if self._autotools:
             return self._autotools
+        if (self.settings.compiler == "apple-clang" and
+            tools.Version(self.settings.compiler.version) == "12" and
+            self.version == "1.7.0"):
+            with tools.chdir( self._source_subfolder ):
+                os.remove( "configure" )
+                self.run( "./buildconf" )
```

Comment:
2 suggestions:
- add libtool to build requirements()
- call autoreconf in build()

But it will make recipe confusing, with conditional build_requirements for apple-cang 12 only. In the meantime we can't unconditionally call autoreconf currently, because there is patch in configure for MinGW.
So maybe the better solution right now is to patch configure.

Reply:
I submitted #3985 for building with mingw@Linux.
The problem of a failed build is still present:
```
tools/gen_test_char.exe > include/private/apr_escape_test_char.h
/bin/sh: line 1: tools/gen_test_char.exe: cannot execute binary file: Exec format error
```

Reply:
Thank you for the suggestions.
I will add the requirement for libtool.

Regarding calling autoreconf. Keep in mind it was recommended to use buildconf and not autoreconf. I'm fine with using either and am willing to change this to autoreconf. But wanted to throw this note about buildconf here before going ahead.

Reply:
> 2 suggestions:
> 
> * add libtool to build requirements()

Hmmm, another question. I was about to add that build_requirement, but would it not make sense to also add:
* automake/1.16.3
* autoconf/2.71

then?

Reply:
it comes with libtool for free, and build requirements can't be overridden, so unfortunately it doesn't scale to add libtool, automake and autoconf explicitly in build_requirements.